Classes/Repository/RepositoryViewController.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  RepositoryViewController.swift
+//  Freetime
+//
+//  Created by Sherlock, James on 29/07/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+import IGListKit
+
+class RepositoryViewController: UIViewController,
+                                FeedDelegate,
+                                ListAdapterDataSource,
+                                SegmentedControlSectionControllerDelegate,
+                                SearchLoadMoreSectionControllerDelegate {
+
+    private let client: RepositoryClient
+    private lazy var feed: Feed = { Feed(viewController: self, delegate: self) }()
+    private let selection: SegmentedControlModel
+    
+    private let noIssuesResultsKey = "noIssuesResultsKey" as ListDiffable
+    private let noPullRequestsResultsKey = "noPullRequestsResultsKey" as ListDiffable
+    private let loadMore = "loadMore" as ListDiffable
+    
+    init(client: GithubClient, repo: RepositoryLoadable) {
+        self.client = RepositoryClient(githubClient: client, repo: repo)
+        self.selection = SegmentedControlModel.forRepository(repo)
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        feed.viewDidLoad()
+        feed.adapter.dataSource = self
+        title = client.repo.nameWithOwner
+    }
+
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        feed.viewWillLayoutSubviews(view: view)
+    }
+    
+    // MARK: Data Loading/Paging
+    
+    private func update(dismissRefresh: Bool, animated: Bool = true) {
+        feed.finishLoading(dismissRefresh: dismissRefresh, animated: animated)
+    }
+    
+    func reload() {
+        client.load(containerWidth: view.bounds.width) {
+            self.update(dismissRefresh: true, animated: true)
+        }
+    }
+    
+    func loadNextPage() {
+        if selection.issuesSelected {
+            client.loadMoreIssues(containerWidth: view.bounds.width) {
+                self.update(dismissRefresh: true, animated: false)
```

Comment:
This _could_ leak and thus you should probably not retain `self` here? I am personally a fan of `unowned let` to make sure I crash if this block is called when it shouldn't (`client` should never outlive `self`). Ryan mostly defaults to `[weak self]` that is probably best? 🤔